src/training/calculatorreview/Main.java

```diff
@@ -4,9 +4,79 @@
 
 public class Main {
 
+    private static Scanner in;
+
     public static void main(String[] args) {
+        printWelcomeMessage();
+
+        in = new Scanner(System.in);
+
+        while (true) {
+            performOneCalculation();
+        }
+    }
+
+    private static void performOneCalculation() {
+        String op = askForOperator();
+        int[] n = askForNumberArray(op);
+        int a = calculateAnswer(op, n);
+
+        System.out.println("The answer is: " + a);
+        System.out.println();
+    }
+
+    private static void printWelcomeMessage() {
         System.out.println("Welcome to the calculator!");
         System.out.println("==========================");
-        new Scanner(System.in);
+    }
+
+    private static String askForOperator() {
+        String[] allowedOperators = { "+", "-", "*", "/" };
+        System.out.print("Please enter the operator: ");
+        String op = in.nextLine();
+        return op;
+    }
+
+    private static int[] askForNumberArray(String op) {
+        System.out.print("How many numbers do you want to " + op + "? ");
+        int count = Integer.parseInt(in.nextLine());
+        int numbers[] = new int[]{count};

```

Comment:
Please can you check this syntax? I haven't used it before, but I think this might return `count `as an entry in the array- I think you need `new int[count]` to declare an array of size `count`